docs/benchmark.md

```diff
@@ -12,86 +12,202 @@ It's important to look at the units than at the percentages. In general, RxDogTa
-| RxDogTagAndroidPerf.flowable_true_subscribe_simple | 12,725ns | 0.013ms | 8556.46%
+|----------|------------|-----------|------------------|
+| RxDogTagAndroidPerf.flowable_simple[enabled=false,times=0,guardedDelegateEnabled=false]            |         350ns |   0.000ms |
+| RxDogTagAndroidPerf.flowable_simple[enabled=true,times=0,guardedDelegateEnabled=false]             |      14,616ns |   0.015ms | 4076.00% |
 
 #### Complex (Observable)
 | Benchmark | Time (ns) | Time (ms) | Percent Increase |
-|---------- |-----------|-----------|------------------|
-| RxDogTagAndroidPerf.observable_false_subscribe_complex | 5,322ns | 0.005ms
-| RxDogTagAndroidPerf.observable_true_subscribe_complex | 25,046ns | 0.025ms | 370.61%
+|----------|------------|-----------|------------------|
+| RxDogTagAndroidPerf.observable_complex[enabled=false,times=0,guardedDelegateEnabled=false]         |       9,485ns |  0.009ms |
+| RxDogTagAndroidPerf.observable_complex[enabled=true,times=0,guardedDelegateEnabled=false]          |      27,028ns |  0.027ms | 184.96% |
 
 #### Complex (Flowable)
 | Benchmark | Time (ns) | Time (ms) | Percent Increase |
-|---------- |-----------|-----------|------------------|
-| RxDogTagAndroidPerf.flowable_false_subscribe_complex | 8,376ns | 0.008ms
-| RxDogTagAndroidPerf.flowable_true_subscribe_complex | 49,184ns | 0.049ms | 487.20%
+|----------|------------|-----------|------------------|
+| RxDogTagAndroidPerf.flowable_complex[enabled=false,times=0,guardedDelegateEnabled=false]           |      15,904ns |  0.016ms |
+| RxDogTagAndroidPerf.flowable_complex[enabled=true,times=0,guardedDelegateEnabled=false]            |      50,602ns |  0.051ms | 218.17% |
 
-### End-to-end amortized cost
+
+### E2E amortized cost
 
 This measures the end-to-end amortized cost.
 
 #### Observable
 | Benchmark | Time (ns) | Time (ms) | Percent Increase |
-|---------- |-----------|-----------|------------------|
-| RxDogTagAndroidPerf.observable_false_e2e | 99,010ns | 0.099ms
-| RxDogTagAndroidPerf.observable_true_e2e | 145,833ns | 0.146ms | 47.29%
+|----------|------------|-----------|------------------|
+| RxDogTagAndroidPerf.observable_e2e[enabled=false,times=0,guardedDelegateEnabled=false]             |      76,363ns |     0.076ms |
+| RxDogTagAndroidPerf.observable_e2e[enabled=false,times=1,guardedDelegateEnabled=false]             |      78,558ns |     0.079ms |    2.87% |
+| RxDogTagAndroidPerf.observable_e2e[enabled=true,times=1,guardedDelegateEnabled=false]              |     119,878ns |     0.120ms |   56.98% |
+| RxDogTagAndroidPerf.observable_e2e[enabled=true,times=1,guardedDelegateEnabled=true]               |     126,007ns |     0.126ms |   65.01% |
+| RxDogTagAndroidPerf.observable_e2e[enabled=true,times=0,guardedDelegateEnabled=false]              |     135,660ns |     0.136ms |   77.65% |
+| RxDogTagAndroidPerf.observable_e2e[enabled=false,times=1,000,guardedDelegateEnabled=false]         |     320,312ns |     0.320ms |  319.46% |
```

Comment:
Is it possible to break the E2E cost by number of items? For example it's not valuable to compare e2e cost of 1000 item Observable with RxDogTag disabled with 1 item Observable with RxDogTag enabled. Giving comparisons for same number of items will be more useful since the % change will make more sense there. 

Right now you have to do a lot of cognitive calculations to know what the real e2e cost of RxDogTag is since you subtract the 1000 Observable(with RxDogTag) with 1000 Observable(without RxDogTag)